### PR TITLE
Fixing train.py to lim's code

### PR DESCRIPTION
**Pull request by GreatSnoopyMe**
## Description
Fix train.py traning_default.yaml and util.py to fit the lim's code.

## How Has This Been Tested?
I have run it on the google cloud. All datasets can be run.

### CODE REVIEW

File: benchmarks/NodeClassification/utils.py [hunk 1]
```diff
@@ -30,6 +30,8 @@
 from models.linkx import LINKX
 
 Models_need_to_be_densed = ["GCN", "GraphSAGE", "GAT", "MixHop", "LINKX"]
+Datasets_need_to_be_undirected = ["pokec", "genius", "penn94", "twitch-gamers"]
+Models_using_adamW = ["LINKX", "MixHop"]
```
Comment (jiaqima) [inline benchmarks/NodeClassification/utils.py:34]: Remove this line

File: benchmarks/NodeClassification/train.py [hunk 2]
```diff
@@ -16,7 +16,8 @@
 from utils import generate_model, parse_args, Models_need_to_be_densed,\
                   load_config_file, check_multiple_split,\
                   EarlyStopping, set_seed, check_binary_classification,\
-                  eval_rocauc
+                  eval_rocauc, Datasets_need_to_be_undirected,\
+                  Models_using_adamW
```
Comment (jiaqima) [inline benchmarks/NodeClassification/train.py:20]: Remove Models_using_adamW. Create a separate train yaml file for LINKX to use AdamW. The original paper of MixHop did not use AdamW so there is no need to use AdamW for MixHop.

**Review by jiaqima — CHANGES_REQUESTED** [review]

**Comment by StevenGolden1203** [comment]
I think AdamW should be a case of optimizer. For example, in `train_default.yaml`, we need to add:
```
optimizer: adam
```
For the models need to use AdamW, we will have:
```
optimizer: adamW
```
in their own `train.yaml` file. `use_adamW` May be removed

File: benchmarks/NodeClassification/train.py (hunk 2 above)
Comment (jiaqima) [reply benchmarks/NodeClassification/train.py:20]: The separate train yaml file could be named as `LINKX_train.yaml`

Comment (GreatSnoopyMe) [reply benchmarks/NodeClassification/train.py:20]: I think they do use AdamW for Mixhop. I have read the code and check it. Maybe I have mistaken it? 
They did not specify that the Mixhop should not use adam in experiment/mixhop_exp.sh

File: benchmarks/NodeClassification/train.py [hunk 3]
```diff
@@ -127,15 +126,16 @@ def main():
...
         optimizer = torch.optim.AdamW(
             model.parameters(), lr=train_cfg["lr"],
             weight_decay=train_cfg["weight_decay"])
-    else:
+    elif train_cfg["optimizer"] == "Adam":
         optimizer = torch.optim.Adam(
             model.parameters(), lr=train_cfg["lr"],
             weight_decay=train_cfg["weight_decay"])
+    else:
+        raise NotImplementedError
```
Comment (jiaqima) [inline benchmarks/NodeClassification/train.py:138]: NotImplementedError(f"Optimizer {train_cfg['optimizer']} is not supported.")

**Review by jiaqima — APPROVED** [review]
Left a minor comment. Otherwise LGTM

**Pull request merged**